src/main/rx/cc2500_sfhss.c

```diff
@@ -81,47 +81,59 @@ static timeMs_t timeTunedMs;
 static int8_t bindOffset_max = 0;
 static int8_t bindOffset_min = 0;
 
+const cc2500RegisterConfigElement_t cc2500SfhssConfigPart1[] =
+{
+    { CC2500_02_IOCFG0,   0x01 },
+    { CC2500_03_FIFOTHR,  0x07 },
+    { CC2500_04_SYNC1,    0xD3 },
+    { CC2500_05_SYNC0,    0x91 },
+    { CC2500_06_PKTLEN,   0x0D },
+    { CC2500_07_PKTCTRL1, 0x04 },
+    { CC2500_08_PKTCTRL0, 0x0C },
+    { CC2500_09_ADDR,     0x29 },
+    { CC2500_0B_FSCTRL1,  0x06 }
+};
+
+const cc2500RegisterConfigElement_t cc2500SfhssConfigPart2[] =
+{
+    { CC2500_0D_FREQ2,    0x5C },
+    { CC2500_0E_FREQ1,    0x4E },
+    { CC2500_0F_FREQ0,    0xC4 },
+    { CC2500_10_MDMCFG4,  0x7C },
+    { CC2500_11_MDMCFG3,  0x43 },
+    { CC2500_12_MDMCFG2,  0x03 },
+    { CC2500_13_MDMCFG1,  0x23 },
+    { CC2500_14_MDMCFG0,  0x3B },
+    { CC2500_15_DEVIATN,  0x44 },
+    { CC2500_17_MCSM1,    0x0F },
+    { CC2500_18_MCSM0,    0x08 },
+    { CC2500_19_FOCCFG,   0x1D },
+    { CC2500_1A_BSCFG,    0x6C },
+    { CC2500_1B_AGCCTRL2, 0x03 },
+    { CC2500_1C_AGCCTRL1, 0x40 },
+    { CC2500_1D_AGCCTRL0, 0x91 },
+    { CC2500_21_FREND1,   0x56 },
+    { CC2500_22_FREND0,   0x10 },
+    { CC2500_23_FSCAL3,   0xA9 },
+    { CC2500_24_FSCAL2,   0x0A },
+    { CC2500_25_FSCAL1,   0x00 },
+    { CC2500_26_FSCAL0,   0x11 },
+    { CC2500_29_FSTEST,   0x59 },
+    { CC2500_2C_TEST2,    0x88 },
+    { CC2500_2D_TEST1,    0x31 },
+    { CC2500_2E_TEST0,    0x0B },
+    { CC2500_3E_PATABLE,  0xFF }
+};
+
 static void initialise()
 {
     cc2500Reset();
 
-    cc2500WriteReg(CC2500_02_IOCFG0,   0x01);
-    cc2500WriteReg(CC2500_03_FIFOTHR,  0x07);
-    cc2500WriteReg(CC2500_04_SYNC1,    0xD3);
-    cc2500WriteReg(CC2500_05_SYNC0,    0x91);
-    cc2500WriteReg(CC2500_06_PKTLEN,   0x0D);
-    cc2500WriteReg(CC2500_07_PKTCTRL1, 0x04);
-    cc2500WriteReg(CC2500_08_PKTCTRL0, 0x0C);
-    cc2500WriteReg(CC2500_09_ADDR,     0x29);
-    cc2500WriteReg(CC2500_0B_FSCTRL1,  0x06);
+    cc2500ApplyRegisterConfig(cc2500SfhssConfigPart1, sizeof(cc2500SfhssConfigPart1));
+
     cc2500WriteReg(CC2500_0C_FSCTRL0,  rxCc2500SpiConfig()->bindOffset);
```

Comment:
I have no idea if the ordering of this register write is critical so I left it how it was and split the hardcoded config to a part 1/2. If it turns out that we can write this register at the end with no impact then we could save a few more bytes by combining the sections.

Reply:
I'd leave it as it is - the code that we have was probably created by reverse engineering SFHSS RX firmware, and we may never know the 'why' of how it is the way it is.